### PR TITLE
Added Postman Sort in JavaScript

### LINKED ISSUE #2251: Postman sort
- [ ] C - @jhasaurabh312
- [x] C++ - @coderprasukj
- [x] C#
- [ ] CoffeeScript
- [ ] Dart - @Rukmini-Meda
- [ ] Go
- [ ] Java - @bluejay229
- [x] JS - @coderprasukj
- [ ] Kotlin - @myselfanuj
- [ ] Php
- [ ] Python - @jhasaurabh312
- [x] Readme - @shwetagurnani
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by coderprasukj**
Fixes #2251 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Postman Sort Implementation in JavaScript

#### Languages Used:
- JavaScript

#### Files Added:
- Postman_Sort.js

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by coderprasukj** [comment]
@jainaman224 @asha15 @SakshiShreya I have created a PR for Postman Sort in JavaScript. Kindly have a look at it.

**Comment by coderprasukj** [comment]
@somya-kapoor I have created a PR for Postman Sort in JavaScript. Kindly have a look at it.

File: Postman_Sort/Postman_Sort.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+let temp, max, count, maxdigits = 0, c = 0;
+let t1, t2, k, t, n = 1, p, s;
+
+/* Function to arrange the of sequence having same base */
+function arrange(p, s) {
+    for(var i = p; i < s - 1; i++) {
+        for(var j = i + 1; j < s; j++) {
+            if(arr1[i] > arr1[j]) {
+                // swap array1[i] and array1[j]
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:13]: You are swapping numbers at so many places. Please write a common swap function.

File: Postman_Sort/Postman_Sort.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+let arr1 = new Array();
+for(let i = 0; i < count; i++) {
+    // first element in t
+    t = arr[i];
+    // Calculating the Number of digits using logarithmic
+    c = Math.trunc(Math.log(arr[i]/Math.log(10)));
+    // number of digits in a each number
+    if(maxdigits < c)
+    maxdigits = c;
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:40]: Please add 4 space indentation.

File: Postman_Sort/Postman_Sort.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+while(--maxdigits)
+n = n * 10;
+
+for(var i = 0; i < count; i++) {
+    // MSB - Dividing by particular base
+    max = arr[i] / n; 
+    t = i;
+    for(var j = i + 1; j < count; j++) {
+        if(max > arr[j]/n) {
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:51]: Please add space around /

File: Postman_Sort/Postman_Sort.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            max = arr[j] / n;
+            t = j;
+        }
+    }
+    temp = arr1[t];
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:56]: Add a swap function.

File: Postman_Sort/Postman_Sort.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    temp = arr[t];
+    arr[t] = arr[i];
+    arr[i] = temp;
+}
+
+while (n >= 1) {
+    for(var i = 0; i < count;) {
+        t1 = arr[i] / n;
+        for(j = i + 1; t1 == (arr[j] / n); j++);
+        arrange(i, j);
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:68]: Please add 4 space indentation.

File: Postman_Sort/Postman_Sort.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+let arr = [70, 80, 20, 50, 10, 60, 40, 100, 90, 30];
+console.log("The Unsorted Array is : ");
+console.log(arr);
+
+let arr1 = new Array();
+for(let i = 0; i < count; i++) {
+    // first element in t
+    t = arr[i];
+    // Calculating the Number of digits using logarithmic
+    c = Math.trunc(Math.log(arr[i]/Math.log(10)));
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:37]: Space around /

File: Postman_Sort/Postman_Sort.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    // number of digits in a each number
+    if(maxdigits < c)
+    maxdigits = c;                               
+}
+
+while(--maxdigits)
+n = n * 10;
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:44]: Please add 4 space indentation.

File: Postman_Sort/Postman_Sort.js (hunk 4 above)
Comment (coderprasukj) [reply Postman_Sort/Postman_Sort.js:56]: in Javascript, Call by reference functions don't work as there are no pointers in js. Therefore i haven't used swap function here.

**Comment by coderprasukj** [comment]
@jainaman224 @singh-shreya6 @somya-kapoor I have modified the code. Kindly have a look at it.

File: Postman_Sort/Postman_Sort.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+let arr1 = new Array();
+for(let i = 0; i < count; i++) {
+    // first element in t
+    t = arr[i];
+    // Calculating the Number of digits using logarithmic
+    c = Math.trunc(Math.log(arr[i] / Math.log(10)));
+    // number of digits in a each number
+    if(maxdigits < c)
+    maxdigits = c;
```
Comment (singh-shreya6) [inline Postman_Sort/Postman_Sort.js:38]: This statement should have indentation of 4 spaces after if

File: Postman_Sort/Postman_Sort.js (hunk 4 above)
Comment (singh-shreya6) [reply Postman_Sort/Postman_Sort.js:54]: ok

**Comment by coderprasukj** [comment]
@singh-shreya6 I have done the required changes. Please have a look at it.

**Comment by singh-shreya6** [comment]
Please squash your commits now, so that it can be merged

**Comment by coderprasukj** [comment]
@singh-shreya6  I have squashed the commits. Kindly do have a look at it.

**Pull request merged**